File: cloud-functions/apiKeys/exchanges/sessionFactory.js
const BinanceSession = require("./binance/session");
const BingXSession = require("./bingx/session");
const BybitSession = require("./bybit/session");
const KuCoinSession = require("./kucoin/session");
const TestnetSession = require("./testnet/session");
const CustomError = require("../utils/error");

/**
 * Factory function to create an exchange session.
 * @param {string} exchange - The name of the exchange ("kucoin", "binance", "bingx", "testnet").
 * @param {string} apiKey - API key for the session.
 * @param {string} apiSecret - API secret for the session.
 * @param {string} [apiPassphrase] - API passphrase for the session (optional, only for KuCoin).
 * @returns {Object} An instance of the relevant session class.
 * @throws {CustomError} Throws a custom error if the exchange is unknown.
 */
function createSession(exchange, apiKey, apiSecret, apiPassphrase) {
  switch (exchange) {
    case "binance":
      return new BinanceSession(apiKey, apiSecret);
    case "bingx":
      return new BingXSession(apiKey, apiSecret);
    case "bybit":
      return new BybitSession(apiKey, apiSecret);
    case "kucoin":
      return new KuCoinSession(apiKey, apiSecret, apiPassphrase);
    case "testnet":
      return new TestnetSession(apiKey, apiSecret);

    default:
      throw new CustomError({
        message: "Unknown Exchange",
        status: 400,
        source: "createSession",
      });
  }
}

module.exports = createSession;

File: cloud-functions/submitTrade-py/trade/session_factory.py
from exchanges.bingx.api.session import BingXSession
from exchanges.bybit.api.session import BybitSession
from exchanges.binance.api.session import BinanceSession
from exchanges.kucoin.api.session import KucoinSession


def create_session(exchange, api_key, api_secret, api_passphrase=None):
    if exchange == "kucoin":
        return KucoinSession(api_key, api_secret, api_passphrase)
        pass
    elif exchange == "binance":
        return BinanceSession(api_key, api_secret)
        pass
    elif exchange == "bingx":
        return BingXSession(api_key, api_secret)
    elif exchange == "bybit":
        return BybitSession(api_key, api_secret)
    elif exchange == "testnet":
        # return TestnetSession(api_key, api_secret)
        pass
    else:
        return
